lib/rubocop/cop/rspec/single_argument_message_chain.rb

```diff
@@ -39,12 +37,27 @@ def autocorrect(node)
             corrector.replace(node.loc.selector, replacement(node.method_name))
             message_chain(node) do |arg|
               autocorrect_hash_arg(corrector, arg) if single_key_hash?(arg)
+              autocorrect_array_arg(corrector, arg) if arg.array_type?
             end
           end
         end
 
         private
 
+        def valid_usage?(node)
+          return true unless node.literal? || node.array_type?
```

Comment:
Isn’t Array also a literal?

Reply:
Nope. I guess it's because it may contain non-literals. I'm checking Rubocop's Node source every time.

Reply:
Hmm, `RuboCop::AST::Node::TRUTHY_LITERALS` includes `:array`, and with https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/ast/node.rb#L28-L32 and https://github.com/rubocop-hq/rubocop/blob/40d81efc59d28038a3f64b02f7ff6f324d8667ce/lib/rubocop/ast/node.rb#L374-L376 it looks like `#literal?` would always return true for Array nodes.

Reply:
True story. Didn't expected that it's both coposite_literal and a basic one at the same time :)

Reply:
There are literals, basic literals, and composite literals :-)
`BASIC_LITERALS = (LITERALS - COMPOSITE_LITERALS).freeze`

Reply:
Yup, got it. Overlooked it and I'm pretty sure I'm overlooking this for array for at least second time. Might clean it up at some point, but it's not a big deal :)